### PR TITLE
IT-2673: Move scicomp provisioner to GHA

### PR DESCRIPTION
**Pull request by xschildw**
This PR contains a GHA replacement for the existing Travis job.
NOTE: Don't merge before removing .travis.yaml

### CODE REVIEW

File: .yamllint [hunk 1]
```diff
@@ -27,10 +27,3 @@ rules:
   truthy: disable
   new-line-at-end-of-file:
     level: warning
-  quoted-strings:
```
Comment (xschildw) [inline .yamllint:30]: This was not in aws-infra and would have required quoting every string.

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          sceptre_subcommand: 'validate prod'
+  sceptre-deploy:
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/master' }}
+    needs: [ 'pre-commit', 'sceptre-validate' ]
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@master
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@v1
```
Comment (zaro0508) [inline .github/workflows/deploy.yaml:41]: it's v2 now https://github.com/marketplace/actions/configure-aws-credentials-for-github-actions

File: .github/workflows/deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+name: "scicomp-provisioner-gha"
```
Comment (zaro0508) [inline .github/workflows/deploy.yaml:1]: would make more sense if this was same as the filename

**Review by zaro0508 — CHANGES_REQUESTED** [review]
please remove .travis.yml file and remember to clean up any secrets left on travis

**Review by zaro0508 — APPROVED** [review]

File: .github/workflows/deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::055273631518:role/sagebase-github-oidc-sage-ProviderRolescicompprovi-BWN39S0CL6IZ
+          role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          role-duration-seconds: 1800
+      - name: 'Sceptre Launch prod'
+        uses: Sceptre/github-ci-action@master
+        with:
+          sceptre_version: '4.0.2'
+          sceptre_subcommand: 'launch -y prod'
```
Comment (zaro0508) [inline .github/workflows/deploy.yaml:37]: we also want the `--prune` flag, https://github.com/Sage-Bionetworks/scicomp-provisioner/blob/master/.travis.yml#L30

File: .github/workflows/deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::055273631518:role/sagebase-github-oidc-sage-ProviderRolescicompprovi-BWN39S0CL6IZ
+          role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          role-duration-seconds: 1800
+      - name: 'Sceptre Launch prod'
+        uses: Sceptre/github-ci-action@master
+        with:
```
Comment (zaro0508) [inline .github/workflows/deploy.yaml:35]: we also need to install the sceptre-ssm-resolver, https://github.com/Sage-Bionetworks/scicomp-provisioner/blob/master/.travis.yml#L14

File: .github/workflows/deploy.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+          python-version: 3.9
+      - uses: pre-commit/action@v3.0.0
+  sceptre-deploy:
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/master' }}
+    needs: [ 'pre-commit' ]
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@master
```
Comment (zaro0508) [inline .github/workflows/deploy.yaml:25]: we should use v3, not master   https://github.com/marketplace/actions/checkout

**Review by zaro0508 — CHANGES_REQUESTED** [review]

**Review by zaro0508 — APPROVED** [review]
looks good, just one minor update and it's good to go.

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**